go/ai/retriever.go

```diff
@@ -55,58 +71,154 @@ type RetrieverResponse struct {
 
 // DefineIndexer registers the given index function as an action, and returns an
 // [Indexer] that runs it.
-func DefineIndexer(provider, name string, index func(context.Context, *IndexerRequest) error) *Indexer {
+func DefineIndexer(provider, name string, index func(context.Context, *IndexerRequest) error) Indexer {
 	f := func(ctx context.Context, req *IndexerRequest) (struct{}, error) {
 		return struct{}{}, index(ctx, req)
 	}
-	return (*Indexer)(core.DefineAction(provider, name, atype.Indexer, nil, f))
+	return (*IndexerAction)(core.DefineAction(provider, name, atype.Indexer, nil, f))
 }
 
-// IsDefinedIndexer reports whether an [Indexer] is defined.
+// IsDefinedIndexer reports whether an [IndexerAction] is defined.
```

Comment:
I would say "Indexer" here and throughout. I don't think the user needs to know about actions to use this API.

Reply:
I noticed the same for generate.go (Models). ModelAction never actually appears in the public API.

Reply:
Done.